Bottom: f654721cb332321b77969cbcbbeb7e0dc8190e83
Top:    07715f1e2de4a525748a867233cc3706e9984d65
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:57 -0500

Add subtree support scripts


---
diff --git a/scripts/subtree-extras.sh b/scripts/subtree-extras.sh
new file mode 100755
index 0000000..8cab898
--- /dev/null
+++ b/scripts/subtree-extras.sh
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -e
+
+extras="\
+docker, docker-exim4                                 , AndrewCooper , git@github.com:AndrewCooper/docker-exim4.git                                                   , origin/master, master
+docker, docker-mediawiki                             , AndrewCooper , git@github.com:AndrewCooper/docker-mediawiki.git                                               , origin/master, downgrade-extensions
+docker, docker-openldap                              , AndrewCooper , git@github.com:AndrewCooper/docker-openldap.git                                                , origin/master, master
+docker, docker-openssl                               , AndrewCooper , git@github.com:AndrewCooper/docker-openssl.git                                                 , origin/master, master
+docker, docker-svn                                   , AndrewCooper , git@github.com:AndrewCooper/docker-svn                                                         , origin/master, master
+docker, docker-SupportSite_http_server               , andrew.cooper, git@git.novatech-llc.com:andrew.cooper/docker-SupportSite_http_server.git                      , origin/master, apache-2.4.7
+docker, docker-TestStation_http_server               , andrew.cooper, git@git.novatech-llc.com:andrew.cooper/docker-TestStation_http_server.git                      , origin/master, apache-2.4.7
+docker, docker-TestStation_SupportSite_backup_restore, andrew.cooper, git@git.novatech-llc.com:andrew.cooper/docker-TestStation_SupportSite_backup_restore.git       , origin/master, limit-backup-dbs
+"
+
+git tag -f pre-extras
+git clean -dxff
+while IFS=", " read prefix name remote repo frombranch branch; do
+    if [ "x$name" == "x" ]; then continue; fi
+
+    fullremote="${name}/${remote}"
+
+    if [ "x$prefix" == "x" ]; then prefix="${name}"; else prefix="${prefix}/${name}"; fi 
+    if git remote get-url "${fullremote}" 2>/dev/null; then rmtcmd="set-url"; else cmd="add"; fi
+
+    git remote ${rmtcmd} "${fullremote}" "${repo}"
+    git fetch "${fullremote}" "${branch}" > /dev/null
+
+    rev_list="$( 
+        git rev-list --reverse ${name}/${frombranch}..${fullremote}/${branch}
+    )"
+
+    for r in $rev_list; do
+        git show --oneline --numstat $r
+        git cherry-pick --strategy recursive -X subtree=${prefix} $r
+    done
+
+done <<< $extras
diff --git a/scripts/subtree-import.sh b/scripts/subtree-import.sh
index 965738f..3ac12ba 100755
--- a/scripts/subtree-import.sh
+++ b/scripts/subtree-import.sh
@@ -49,28 +49,29 @@ docker, docker-TestStation_tftp_data                 , origin       , git@git.no
 docker, docker-TestStation_tftp_server               , origin       , git@git.novatech-llc.com:NovaTech-Testing/docker-TestStation_tftp_server.git                   , master
 "
 git tag -f 'pre-import'
-git clean -dxff
-while IFS=", " read prefix name remote repo branch; do
-    if [ "x$name" == "x" ]; then continue; fi
+# git clean -dxff
+prefix="$1";shift
+name="$1";shift
+remote="$1";shift
+repo="$1";shift
+branch="$1";shift
 
-    fullremote="${name}/${remote}"
+fullremote="${name}/${remote}"
 
-    message="${name}: Merge '${fullremote}' at ${branch}"
-    rmtcmd="none"
-    cmd="none"
+message="${name}: Merge '${fullremote}' at ${branch}"
+rmtcmd="none"
+stcmd="none"
 
-    if [ "x$prefix" == "x" ]; then prefix="${name}"; else prefix="${prefix}/${name}"; fi 
-    if git remote get-url "${fullremote}" 2>/dev/null; then rmtcmd="set-url"; else cmd="add"; fi
-    if [ -d "${prefix}" ]; then cmd="pull"; else cmd="add"; fi
+if [ "x$prefix" == "x" ]; then prefix="${name}"; else prefix="${prefix}/${name}"; fi 
+if git remote get-url "${fullremote}" 2>/dev/null; then rmtcmd="set-url"; else rmtcmd="add"; fi
+if [ -d "${prefix}" ]; then stcmd="pull"; else stcmd="add"; fi
 
-    echo ${message}
-    set -x
-    git config --add subtree."$prefix".remote "${fullremote}"
-    git remote ${rmtcmd} "${fullremote}" "${repo}"
-    git fetch "${fullremote}" "${branch}" > /dev/null
-    git subtree ${cmd} --prefix="${prefix}" --message="${message}" "${fullremote}" "${branch}" > /dev/null
-    { set +x; } 2> /dev/null
-    echo
-    echo
-
-done <<< $sources
+echo ${message}
+set -x
+git config --add subtree."$prefix".remote "${fullremote}"
+git remote ${rmtcmd} "${fullremote}" "${repo}"
+git fetch "${fullremote}" "${branch}" > /dev/null
+git subtree ${stcmd} --prefix="${prefix}" --message="${message}" "${fullremote}" "${branch}" > /dev/null
+{ set +x; } 2> /dev/null
+echo
+echo
diff --git a/scripts/subtree-push.sh b/scripts/subtree-push.sh
new file mode 100755
index 0000000..7d680bf
--- /dev/null
+++ b/scripts/subtree-push.sh
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -e
+
+for st in $*; do
+    git subtree split --prefix="$st" --branch="$st/master"
+    git push -f "$st/AndrewCooper" "$st/master":master
+done
